diffsync/__init__.py

```diff
@@ -470,8 +471,10 @@ def sync_from(
             flags (DiffSyncFlags): Flags influencing the behavior of this sync.
             callback (function): Function with parameters (stage, current, total), to be called at intervals as the
                 calculation of the diff and subsequent sync proceed.
+            diff (Diff): An existing diff to be used rather than generating a completely new diff.
         """
-        diff = self.diff_from(source, diff_class=diff_class, flags=flags, callback=callback)
+        if not diff:
+            diff = self.diff_from(source, diff_class=diff_class, flags=flags, callback=callback)
```

Comment:
If both `diff` and `diff_class` parameters are non-None, should we assert that `isinstance(diff, diff_class)`? Or should we disallow specifying `diff_class` at all if `diff` was specified?

Reply:
Good catch! I think it would make sense to compare and raise an error if they don't match. The reason being is that we want to make sure the user gets their intended synchronization by making sure the diff they passed in is subclassed by the diff_class that is passed in or that is default.